jazz_assets/index.js

```diff
@@ -0,0 +1,265 @@
+            reject({
+                result: "inputError",
+                message: "Asset data is required for updating an asset"
+            });
+        }
+    
+        if (event.method === "POST" && Object.keys(event.body).length === 0 && Object.keys(event.path).length === 0) {
+            reject({
+                result: "inputError",
+                message: "Asset details are required for creating an asset"
+            });
+        }
+    
+        if (event.method === "POST" && Object.keys(event.path).length > 0 && event.path.id !== "search") {
+            reject({
+                result: "inputError",
+                message: "Parameters are not supported"
+            });
+        }
+        resolve();
+    });
+};
+
+function validateUpdateData(assets_id, update_data) {
+    logger.debug("Inside validateUpdateData");
+    return new Promise((resolve, reject) => {
+        validateutils.validateUpdatePayload(assets_id, update_data, (error, data) => {
+            if(error) {
+                reject(error);
+            } else {
+                resolve(data);
+            }
+        });
+    });
+};
+
+function updateAssetsData(assets_id, update_data) {
+    logger.debug("Inside updateAssetsData");
+    return new Promise((resolve, reject) => {
+        if(Object.keys(update_data).length !== 0) {
+            crud.update(assets_id, update_data, (error, data) => {
+                if(error) {
+                    reject(error);
+                } else {
+                    resolve(data);
+                }
+            });
+        } else {
+            reject({result:"inputError", message:"Provided Asset data can not be updated."});
+        }
+    });
+};
+
+function validateCreateData(assets_data) {
+    logger.debug("Inside validateCreateData");
+    return new Promise((resolve, reject) => {
+        validateutils.validateCreatePayload(assets_data, (error, data) => {
+            if(error) {
+                reject(error);
+            } else {
+                resolve(data);
+            }
+        });
+    });
+};
+
+function createNewAsset(assets_data) {
+    logger.debug("Inside createNewAsset");
+    return new Promise((resolve, reject) => {
+        crud.create(assets_data, (error, data) => {
+            if(error) {
+                reject(error);
+            } else {
+                resolve(data);
+            }
+        });
+    });
+};
+
+function validateSearchData(assets_data) {
+    logger.debug("Inside validateSearchData");
+    return new Promise((resolve, reject) => {
+        validateutils.validateSearchPayload(assets_data, (error, data) => {
+            if(error) {
+                reject(error);
+            } else {
+                resolve(data);
+            }
+        });
+    });
+};
+
+function postSearch(assets_data) {
+    logger.debug("Inside postSearch");
+    return new Promise((resolve, reject) => {
+        crud.postSearch(assets_data, (error, data) => {
+            if(error) {
```

Comment:
indentation here and other places in the code =?> 'if ('

Reply:
Done